### DISCUSSION TITLE
How to clear version history

### DISCUSSION THREAD

**Discussion by sojusnik** [zim-desktop-wiki/zim-desktop-wiki#2556] [Q&A]
My version history has commits from roughly 10 years and therefore takes ages to open.

What's the best way to clean it, either completely or leave f.i. the last year of commits?

Deleting the `.git` folder in Zim's folder, doesn't work, because it doesn't get recreated.

**Comment by dehesselle** [comment]
Speaking from a git perspective, you can squash commits together. For example, you can squash years 1-9 into a single commit and leave the current year as is. And next year you can repeat this, just with not so many commits as you've already condensed years 1-9 before.
You will have to do this either with git in the command line or possibly with a git GUI client.

This operation cannot be undone later, so be really sure that you want to get rid of your detailed history and/or create a backup just to be safe.

**Reply by sojusnik** [reply]
That's an interesting approach, thanks! Just have to figure out how to do it, since I'm not really a coder myself and have limited experience with git.

**Reply by dehesselle** [reply]
It's way too dangerous for me trying to instruct you remotely, especially if you're unfamiliar with this. Best tip I can give you is to watch several (not just one!) videos on Youtube about squashing so you see different people giving different explanations/presentations about that topic.
